compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingTypes.cpp

```diff
@@ -0,0 +1,120 @@
+// Copyright 2025 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Dialect/Encoding/IR/EncodingTypes.h"
+
+#include "llvm/ADT/SmallVector.h"
+#include "mlir/Dialect/Affine/Utils.h"
+#include "mlir/Dialect/Linalg/Utils/Utils.h"
+#include "mlir/IR/AffineMap.h"
+#include "mlir/IR/BuiltinAttributes.h"
+#include "mlir/IR/BuiltinTypeInterfaces.h"
+
+#include <cassert>
+
+namespace mlir::iree_compiler::IREE::Encoding {
+
+SerializableEncodingAttrInterface
+getSerializableEncodingAttrInterface(RankedTensorType type) {
+  return dyn_cast_or_null<SerializableEncodingAttrInterface>(
+      type.getEncoding());
+}
+
+EncodingAttr getEncodingAttr(RankedTensorType type) {
+  return dyn_cast_or_null<EncodingAttr>(type.getEncoding());
+}
+
+bool hasPackedStorageAttr(RankedTensorType type) {
+  return dyn_cast_or_null<PackedStorageAttr>(type.getEncoding()) != nullptr;
+}
+
+FailureOr<linalg::ContractionDimensions>
+getEncodingContractionDims(EncodingAttr encoding) {
+  auto indexingMapsAttr = encoding.getUserIndexingMaps();
```

Comment:
Would be nice to replace this auto and a few below with concrete types, since the type is not obvious based on the context. But if you don't want to do that in this PR that's fine -- I realize it's mostly code motion.

Reply:
I'm happy to do so, since I touch the code. :)